packages/peregrine/src/hooks/useScrollLock.js

```diff
@@ -0,0 +1,16 @@
+import { useLayoutEffect } from 'react';
+
+/**
+ * A [React Hook]{@link https://reactjs.org/docs/hooks-intro.html} that sets
+ * an attribute on the document element indicating that scrolling should be
+ * locked. This is performed with a layout effect (before paint).
+ *
+ * @kind function
+ *
+ * @param {Boolean} locked Whether scrolling should be locked.
+ */
+export const useScrollLock = locked => {
+    useLayoutEffect(() => {
```

Comment:
Just to clarify, we _must_ use `layoutEffect` here instead of `effect` because we need to do this before paint?

Reply:
Yeah, we know it's going to cause a reflow so I think we want to have it take effect before the next paint.